Notesnook.API/Controllers/MonographsController.cs

```diff
@@ -206,16 +218,61 @@ public async Task<IActionResult> TrackView([FromRoute] string id)
             if (monograph == null) return Content(SVG_PIXEL, "image/svg+xml");
 
             if (monograph.SelfDestruct)
-                await Monographs.DeleteByIdAsync(monograph.Id);
+            {
+                var userId = this.User.FindFirstValue("sub");
+                await Monographs.Collection.UpdateOneAsync(
+                    CreateMonographFilter(userId, new Monograph { ItemId = id }),
+                    Builders<Monograph>.Update
+                    .Set(m => m.Deleted, true)
+                );
+                SyncMonograph(id, null);
+            }
 
             return Content(SVG_PIXEL, "image/svg+xml");
         }
 
         [HttpDelete("{id}")]
-        public async Task<IActionResult> DeleteAsync([FromRoute] string id)
+        public async Task<IActionResult> DeleteAsync([FromQuery] string deviceId, [FromRoute] string id)
         {
-            await Monographs.Collection.DeleteOneAsync(CreateMonographFilter(id));
+            var userId = this.User.FindFirstValue("sub");
+            await Monographs.Collection.UpdateOneAsync(
```

Comment:
Maybe we should `ReplaceOne` instead? The soft deleted monograph should only have the `Deleted` and `ItemId` property. Everything else should be undefined.